Changing voltage calculation to a weighted running average

Erwin-Repolust opened the pull request:
This helps improve precision in the readout and makes it more stable.
The calibration will now not be multiplied by 2 and therefore reflects the change in volts.
Removed check for "invalid voltages" as that just makes it hard for people to check whats going wrong in their setup when they only get "invalid voltage" in the web gui.

blazoncek commented:
@itCarl as you were the last (I think) to update battery usermod, can you verify the code?

itCarl commented:
> @itCarl as you were the last (I think) to update battery usermod, can you verify the code?

Actually i wasn't the last one. Frank aka @softhack007 is the one you should ask.
In my opinion adding smoothing to the voltage makes a lot of sense as @softhack007 and i already discussed here -> #2998.

However, I would suggest that a smoothing calculation should not only be there when working with an esp32 but also in the general voltage calculation formula.

Erwin-Repolust commented:
> > @itCarl as you were the last (I think) to update battery usermod, can you verify the code?
> 
> Actually i wasn't the last one. Frank aka @softhack007 is the one you should ask. In my opinion adding smoothing to the voltage makes a lot of sense as @softhack007 and i already discussed here -> #2998.
> 
> However, I would suggest that a smoothing calculation should not only be there when working with an esp32 but also in the general voltage calculation formula.

Added code and smoothing for the esp8266. The existing code for it was not working properly anyway I believe.
Also there is a new variable now that lets u adjust for different voltage dividers in case u wanna measure something over ~5v.
Changed the filter code to the one mentioned in #2998.
I have not tested the bit for the esp8266 as I don't have access to the hardware but it should work.

3v is used cause of this graph:
![image](https://user-images.githubusercontent.com/50101161/224193545-25214a87-5a3a-45da-a336-4bdb376d4de5.png)

itCarl commented:
Hey 👋,
i just compiled your code Erwin and something is not quite right with the esp8266 code.
I mainly use D1mini's for testing and development and those have a voltage divider built-in.
Wemos D1mini v3.1.0
[Datasheet](https://www.wemos.cc/en/latest/d1/d1_mini_3.1.0.html) -- [Schematic](https://www.wemos.cc/en/latest/_static/files/sch_d1_mini_v3.0.0.pdf)

Wemos D1mini v4.0.0
[Datasheet](https://www.wemos.cc/en/latest/d1/d1_mini.html) -- [Schematic](https://www.wemos.cc/en/latest/_static/files/sch_d1_mini_v4.0.0.pdf)

R1 220k and R2 100k. To get a 'nice' max. of 1v from a 4.2v source we can add a 100k to A0. So that R1 becames 220k + 100k = 320k. When we now apply the voltage divider formula = Vout = (Vin * R2) / (R2 + R1). The result will be something like (4.2v (in) * 100k) / (100k + 320k) = 1v (out). So this is why the current formula multiplies just by maxBatteryVoltage.
Thats why we don't really need the resistor ratio in the voltage calculation formula for the esp8266 inside the usermod. Unless someone is actually using the bare esp8266 chip.
Anyway.
When using the new formula the voltageMultiplyer would be around 1.4 (=4.2v/3.0v) for the resistor setup im currently using.

If i messed something up or understood something wrong. please let me know.

Erwin-Repolust commented:
Ahh thanks for the explanation and for testing, as I said I have no access to the esp8266 right now and from what I could read online it all depends on the specific board you own. There might be no voltage divider at all ~0-1v or one that gives roughly ~0-3.3v and then there is your case where u add a resistor to get ~0-4.2v.
So we just set defaults for both platforms then.
I exposed the new voltage multiplier in the gui. It does help people who use different voltage divider setups in the end.
